cmd/cmd_utils.go

```diff
@@ -0,0 +1,200 @@
+
+						// Prepare full template data
+						var data = map[string]map[string]string{
+							"Arguments": argumentsData,
+							"Flags":     flagsData,
+						}
+
+						// Parse and execute Go templates in the command's steps
+						t, err = template.New(fmt.Sprintf("step-%d", i)).Parse(step)
+						if err != nil {
+							u.PrintErrorToStdErrorAndExit(err)
+						}
+						var tpl bytes.Buffer
+						err = t.Execute(&tpl, data)
+						if err != nil {
+							u.PrintErrorToStdErrorAndExit(err)
+						}
+
+						// Prepare ENV vars
+						var envVarsList []string
+						for _, v := range customEnvVars {
+							key := v.Key
+							val := v.Value
+							valCommand := v.ValueCommand
+
+							if val != "" && valCommand != "" {
+								err = fmt.Errorf("either 'value' or 'valueCommand' can be specified for the ENV var, but not both.\n"+
+									"Custom command '%s %s' defines 'value=%s' and 'valueCommand=%s' for the ENV var '%s'",
+									parentCommand.Name(), commandConfig.Name, val, valCommand, key)
+								u.PrintErrorToStdErrorAndExit(err)
+							}
+
+							// If the command to get the value for the ENV var is provided, execute it
+							if valCommand != "" {
+								valCommandArgs := strings.Fields(valCommand)
+								res, err := e.ExecuteShellCommandAndReturnOutput(valCommandArgs[0], valCommandArgs[1:], ".", nil, false)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								val = res
+							} else {
+								// Parse and execute Go templates in the values of the command's ENV vars
+								t, err = template.New(fmt.Sprintf("env-var-%d", i)).Parse(val)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								var tplEnvVarValue bytes.Buffer
+								err = t.Execute(&tplEnvVarValue, data)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								val = tplEnvVarValue.String()
+							}
+
+							envVarsList = append(envVarsList, fmt.Sprintf("%s=%s", key, val))
+							err = os.Setenv(key, val)
+							if err != nil {
+								u.PrintErrorToStdErrorAndExit(err)
+							}
+						}
+
+						if c.Config.Logs.Verbose {
+							if len(envVarsList) > 0 {
+								u.PrintInfo("\nUsing ENV vars:")
+								for _, v := range envVarsList {
+									fmt.Println(v)
+								}
+							}
+						}
+
+						commandToRun := os.ExpandEnv(tpl.String())
+
+						// Execute the command step
+						stepArgs := strings.Fields(commandToRun)
+						err = e.ExecuteShellCommand(stepArgs[0], stepArgs[1:], ".", envVarsList, false)
+						if err != nil {
+							u.PrintErrorToStdErrorAndExit(err)
+						}
+					}
+				},
+			}
+
+			// Add customCommandFlags
+			for _, flag := range customCommandFlags {
+				if flag.Type == "bool" {
+					if flag.Shorthand != "" {
+						customCommand.PersistentFlags().BoolP(flag.Name, flag.Shorthand, false, flag.Usage)
+					} else {
+						customCommand.PersistentFlags().Bool(flag.Name, false, flag.Usage)
+					}
+				} else {
```

Comment:
If the type is not `bool`, then is this for both integer and string types?

Reply:
yes (for now, we can review it later, but this will work for both string and integer)